contracts/src/common/RegistryDatastore.sol

```diff
@@ -31,7 +31,13 @@ contract RegistryDatastore is IRegistryDatastore {
         view
         returns (address resolver, uint64 expiry, uint32 data)
     {
-        (resolver, expiry, data) = DatastoreUtils.unpack(entries[registry][NameUtils.getCanonicalId(id)].resolverData);
+        address resolverAddr;
+        uint96 flags;
+        (resolverAddr, flags) = DatastoreUtils.unpackWithFlags(entries[registry][NameUtils.getCanonicalId(id)].resolverData);
+        // For backward compatibility, return 0 for expiry and the lower 32 bits of flags for data
```

Comment:
No need to preserve backwards compatibility here, change the interface of RegistryDatastore instead.